### ISSUE TITLE
context's flow_id is None in Environment execute() 

### ISSUE THREAD

**Issue by joeschmid** [PrefectHQ/prefect#2445]
## Description
During testing of `DaskCloudProviderEnvironment` we noticed that `prefect.context.get("flow_id")` returns None in our execute() method. `prefect.context.get("flow_run_id")` does correctly return the flow run id and we are able to retrieve the flow_id with essentially `client.get_flow_run_info(flow_run_id).flow_id`

## Expected Behavior
flow_id should be available in context inside an Environment's execute() method.

## Reproduction
```
def execute(self, storage: "Storage", flow_location: str, **kwargs: Any) -> None:
    flow_id = prefect.context.get("flow_id")
    if not flow_id:  # We've observed cases where flow_id is None
        if not flow_run_info:
            client = Client()
            flow_run_info = client.get_flow_run_info(flow_run_id)
        flow_id = flow_run_info.flow_id
```

## Environment
Docker storage with the Fargate agent. (Agent is running Fargate tasks for the flow runs.)

**Comment by joshmeek** [comment]
Yeah we could easily make this available. When agents create their jobs/processes/tasks/etc. they set the `PREFECT__CONTEXT__FLOW_RUN_ID` for each run deployment. Then when the execute step is called on those jobs it makes a query for flow specific information from the API

https://github.com/PrefectHQ/prefect/blob/b923e6d85770e019e19f95c40df3b4ad822d9002/src/prefect/cli/execute.py#L43-L50

This could easily be extended to grab the flow id and put it into context

**Comment by joeschmid** [comment]
Thanks @joshmeek. I like that idea. As a creator of a new Environment, it seemed logical that if flow_run_id was available in the context during `execute()` that flow_id would also be available.

**Comment by lauralorenz** [comment]
Hey @joeschmid! Us core heads were wondering if what @joshmeek postulated in https://github.com/PrefectHQ/prefect/issues/2445#issuecomment-621216136 is something you'd feel comfortable adding in as part of your DaskCloudProviderEnvironment PR itself https://github.com/PrefectHQ/prefect/pull/2360 or if you still think it is best to do it as a separate issue/PR. Thanks :)

**Comment by joeschmid** [comment]
Ended up making a small, separate PR for this: https://github.com/PrefectHQ/prefect/pull/2461 since it could use a bit of explanation...

For the Docker, Fargate, Local, & Nomad agents it was extremely simple to add this. For the Kubernetes agent and `FargateTaskEnvironment` this wasn't readily available. We could obviously add a GraphQL query for these if we want to keep the behavior uniform, i.e. for all agents/environments to make `flow_id` available in context.

**Comment by cicdw** [comment]
Closed with #2461
